react/features/local-recording/recording/flac/FlacAdapter.js

```diff
@@ -0,0 +1,170 @@
+} from './messageTypes';
+
+const logger = require('jitsi-meet-logger').getLogger(__filename);
+
+/**
+ * Recording adapter that uses libflac in the background
+ */
+export class FlacAdapter extends RecordingAdapter {
+
+    _encoder = null;
+    _audioContext = null;
+    _audioProcessingNode = null;
+    _audioSource = null;
+
+    _stopPromiseResolver = null;
+
+    /**
+     * Implements {@link RecordingAdapter#ensureInitialized}.
+     *
+     * @inheritdoc
+     */
+    ensureInitialized() {
+        if (this._encoder !== null) {
+            return Promise.resolve();
+        }
+
+        const promiseInitWorker = new Promise((resolve, reject) => {
+            // FIXME: workaround for different file names in development/
+            // production environments.
+            // We cannot import flacEncodeWorker as a webpack module,
+            // because it is in a different bundle and should be lazy-loaded
+            // only when flac recording is in use.
+            try {
+                // try load the minified version first
+                this._encoder = new Worker('/libs/flacEncodeWorker.min.js');
+            } catch (exception1) {
+                // if failed, try un minified version
+                try {
+                    this._encoder = new Worker('/libs/flacEncodeWorker.js');
+                } catch (exception2) {
+                    logger.error('Failed to load flacEncodeWorker.');
+                    reject();
+                }
+            }
+
+            // set up listen for messages from the WebWorker
+            this._encoder.onmessage = e => {
+                if (e.data.command === WORKER_BLOB_READY) {
+                    // receiving blob
+                    this._data = e.data.buf;
+                    if (this._stopPromiseResolver !== null) {
+                        this._stopPromiseResolver();
+                        this._stopPromiseResolver = null;
+                    }
+                } else if (e.data.command === DEBUG) {
+                    logger.log(e.data);
+                } else if (e.data.command === WORKER_LIBFLAC_READY) {
+                    logger.debug('libflac is ready.');
+                    resolve();
+                } else {
+                    logger.error(
+                        `Unknown event
+                        from encoder (WebWorker): "${e.data.command}"!`);
+                }
+            };
+
+            this._encoder.postMessage({
+                command: MAIN_THREAD_INIT,
+                config: {
+                    sampleRate: 44100,
+                    bps: 16
+                }
+            });
+        });
+
+        const callbackInitAudioContext = (resolve, reject) => {
+            navigator.getUserMedia(
+
+                // constraints - only audio needed for this app
+                {
+                    audioBitsPerSecond: 44100, // 44 kbps
+                    audio: true,
+                    mimeType: 'application/ogg' // useless?
```

Comment:
Is the ogg mimetype here intentional? "useless?" meaning you just need any string here?